Eugene/MovieSearch/my-app/src/components/MainPage/MainPage.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import "./MainPage.css";
+import Header from "./header/Header";
+import SearchPanel from "./search-panel/SearchPanel";
+import SortPanel from "./sort-panel/SortPanel";
+import IMovie from "../../interface/IMovie/IMovie";
+import MovieList from "./movie-list/MovieList";
+import Footer from "./footer/Footer";
+import { parse, stringify } from "query-string";
+
+import { RouteComponentProps } from "react-router-dom";
+
+interface IMainPageProps extends RouteComponentProps {
+  movies: IMovie[];
+  updateMovies(value: IMovie[]): void;
+}
+
+class MainPage extends React.Component<IMainPageProps> {
+  handleSearch = (search: string, searchBy: string) => {
+    const urlParams = stringify({ search, searchBy });
+    this.props.history.push({
+      pathname: "/search",
+      search: urlParams,
+    });
+  };
```

Comment:
If sortBy is already in url, you will replace it. You have to keep sortBy in URL